### PR TITLE
#7618 - addon.ui commands and structure described

### PR DESCRIPTION
**Pull request by Vegebutcher**
all written from scratch

### CODE REVIEW

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIClickCommand.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+In the following script the robot opens Notepad, then it clicks Close button (**X**), thus closing the Notepad window:
+
+```G1ANT
+program notepad
+ui.click ‴/ui[@name='Untitled - Notepad']/descendant::ui[@name='Close']‴
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UIClickCommand.md]: it's called "Window**s** Tree"

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetRectangleCommand.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+```G1ANT
+program notepad
+ui.getrectangle ‴/ui[@name='Untitled - Notepad']‴
+dialog ♥result
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetRectangleCommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetTextCommand.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+```G1ANT
+program taskmgr
+ui.gettext ‴/ui[@name='Windows Task Manager']/ui[@id='1010']‴
+dialog ♥result
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetTextCommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetFocusCommand.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+program notepad
+program calc
+delay 2
+ui.setfocus ‴/ui[@name='Untitled - Notepad']‴
+keyboard ⋘ALT+F4⋙
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UISetFocusCommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetRootCommand.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+program notepad
+ui.setroot wpath ‴/ui[@name='Untitled - Notepad']‴
+ui.click ‴/ui[@id='MenuBar']/ui[@name='File']‴
+delay 2
+ui.click wpath ‴/descendant::ui[@name='Close']‴
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UISetRootCommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetTextCommand.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+```G1ANT
+program explorer arguments ♥environment⟦USERPROFILE⟧\Desktop
+ui.settext ‴/ui[@name='Desktop']/ui[@id='40965']/descendant::ui[@id='SearchEditBox']‴ text .txt
+```
+
+If you get “*Access denied*” error message when using the `ui.settext` command, run G1ANT.Robot as administrator.
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/UISetTextCommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/WPathStructure.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# wpath
+
+This structure stores WPath information of UI elements and is used by the `ui.` family of commands, `waitfor.ui` included.
+
+The WPath information can be obtained from G1ANT.Robot’s Window Tree panel (select `Window Tree` from `View` menu): expand the tree of a given application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/WPathStructure.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/WaitForUICommand.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+```G1ANT
+program notepad
+waitfor.ui ‴/ui[@name='Untitled - Notepad']/ui[@name='Font']‴
+dialog ‴Font dialog detected!‴
+```
+
+> **Note:** The names of window elements are system language dependent, so scripts with WPaths to these elements can be used only in the same language versions of Windows.
+>
+> To insert a WPath to an element, open a Window Tree panel (select `Window Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `waitfor.ui` command.
```
Comment (WiktoriaPrusik) [inline G1ANT.Addon.UI/G1ANT.Addon.UI/WaitForUICommand.md]: same

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIClickCommand.md (hunk 1 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UIClickCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetRectangleCommand.md (hunk 2 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetRectangleCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetTextCommand.md (hunk 3 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UIGetTextCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetFocusCommand.md (hunk 4 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UISetFocusCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetRootCommand.md (hunk 5 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UISetRootCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/UISetTextCommand.md (hunk 6 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/UISetTextCommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `ui.` command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/WPathStructure.md (hunk 7 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/WPathStructure.md]:
```suggestion
The WPath information can be obtained from G1ANT.Robot’s Windows Tree panel (select `Windows Tree` from `View` menu): expand the tree of a given application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the command.
```

File: G1ANT.Addon.UI/G1ANT.Addon.UI/WaitForUICommand.md (hunk 8 above)
Comment (Vegebutcher) [reply G1ANT.Addon.UI/G1ANT.Addon.UI/WaitForUICommand.md]:
```suggestion
> To insert a WPath to an element, open a Windows Tree panel (select `Windows Tree` from `View` menu) expand the tree of an application window, navigate to a desired element and double-click it. Its WPath will be automatically inserted into the `waitfor.ui` command.
```

**Pull request merged**